pkg/ovsdb/database.go

```diff
@@ -150,32 +159,64 @@ func (con *DatabaseEtcd) PutData(ctx context.Context, key string, obj interface{
 	}
 	return nil
 }
-func (con *DatabaseEtcd) AddMonitor(keysPrefix string, mcr ovsjson.MonitorCondRequest, isV1 bool, hand *handlerKey) {
+
+func (con *DatabaseEtcd) AddMonitors(dbName string, updaters map[string][]*updater, handler handlerKey)  {
+	fmt.Println("DB Monitor 1")
 	con.mu.Lock()
 	defer con.mu.Unlock()
-	if monitor, ok := con.monitors[keysPrefix]; !ok {
-		con.monitors[keysPrefix] = *newMonitor(con.cli, con.prefix+keysPrefix, mcr, isV1, hand)
-	} else {
-		monitor.addHandler(mcr, isV1, hand)
+	fmt.Println("DB Monitor 2")
+	m, ok := con.monitors[dbName]
+	fmt.Printf("DB Monitor 3 %v\n", ok)
+	if !ok {
+		// we don't have monitors for this database yet, create a new one.
+		m = newMonitor(dbName, con)
+		ctxt, cancel := context.WithCancel(context.Background())
+		m.cancel = cancel
+		wch := con.cli.Watch(clientv3.WithRequireLeader(ctxt),con.prefix + dbName,
+			clientv3.WithPrefix(),
+			clientv3.WithCreatedNotify(),
+			clientv3.WithPrevKV())
```

Comment:
this is a code duplication, as you have a specific way to create a monitor (for example with the PrevKV option set) better use a utility function, that way if in future you change options you will only have to modify one location

Reply:
Where do you see "the code duplication" ? only shared option is `clientv3.WithPrefix()` all others are watching related

Reply:
see `AddWatchHandler()`

Reply:
AddWatchHandler is removed